src/Lifecycle/DeferredWriteQueue.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Thunk\Verbs\Lifecycle;
+
+use Thunk\Verbs\Event;
+use Thunk\Verbs\Support\DeferredWriteData;
+use Thunk\Verbs\Support\Wormhole;
+
+class DeferredWriteQueue
+{
+    private array $callbacks = [];
+
+    public function add(Event $event, callable $callback, DeferredWriteData $deferred): void
+    {
+        $class = $deferred->class_name ?? get_class($event);
```

Comment:
I'm thinking the class_name (realistically should be called something else) should default to the event's class, but if there's a unique_by set it should default to that property's class name.